src/App.tsx

```diff
@@ -1,61 +1,106 @@
 import './App.scss';
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { TodoWithUser } from './types/TodoWithUser';
+import { User } from './types/User';
 
 export const App = () => {
+  const todosWithUser = todosFromServer.map(todo => ({
+    ...todo,
+    user: usersFromServer.find(user => (user.id === todo.userId)) as User,
+  }));
+
+  const [visibleTodos, setVisibleTodos] = useState(todosWithUser);
+
+  const [title, setTitle] = useState('');
+  const [titleTouched, setTitleTouched] = useState(false);
+  const hasTitleError = titleTouched && !title;
+
+  const [user, setUser] = useState('0');
+  const [userTouched, setUserTouched] = useState(false);
+  const hasUserError = userTouched && user === '0';
+
+  const clear = () => {
+    setTitle('');
+    setUser('0');
+    setTitleTouched(false);
+    setUserTouched(false);
+  };
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    if (!title || user === '0') {
+      setTitleTouched(true);
+      setUserTouched(true);
+
+      return;
+    }
+
+    const newTodo: TodoWithUser = {
+      id: Math.max(...visibleTodos.map(todo => todo.id)) + 1,
```

Comment:
create function for getting max id and if you use it somewhere else you can extract it to a new file, like service